pkg/argocd/argocd.go

```diff
@@ -58,7 +60,9 @@ func Apply(ctx context.Context, config *rest.Config, namespace, argoImage, redis
 	}
 
 	if err == nil && len(argos.Items) > 0 {
-		return nil
+		// An ArgoCD custom resource exists in our namespace
+		err = fixArgoOperatorDeadlock(ctx, clientset, config, namespace, operatorNamespace)
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("could not fix argocd operator deadlock: %w", err)
```

i usually try to add some context with `fmt.Errorf` to make debugging later errors easier.

also for the errors below.